lib/getConfigGroups.js

```diff
@@ -0,0 +1,77 @@
+/** @typedef {import('./index').Logger} Logger */
+
+import path from 'path'
+
+import { loadConfig } from './loadConfig.js'
+import { ConfigNotFoundError } from './symbols.js'
+import { validateConfig } from './validateConfig.js'
+
+/**
+ * Return matched files grouped by their configuration.
+ *
+ * @param {object} options
+ * @param {Object} [options.configObject] - Explicit config object from the js API
+ * @param {string} [options.configPath] - Explicit path to a config file
+ * @param {string} [options.cwd] - Current working directory
+ * @param {Logger} logger
+ */
+export const getConfigGroups = async ({ configObject, configPath, files }, logger = console) => {
+  // Return explicit config object from js API
+  if (configObject) {
+    const config = validateConfig(configObject, 'config object', logger)
+    return { '': { config, files } }
+  }
+
+  // Use only explicit config path instead of discovering multiple
+  if (configPath) {
+    const { config, filepath } = await loadConfig({ configPath }, logger)
```

Comment:
Should possible errors be handled here?

Reply:
The `loadConfig` contains a try/catch and will return `{}` in case of errors, meaning `config === undefined`. That will be thrown right below on line 31.

Basically this `getConfigGroup` will always throw on errors, so maybe a try/catch should be added here with some better-formatted error logging: 

https://github.com/okonet/lint-staged/blob/cf923b43175b891b7b312592c2c0eb36e7f3d8bd/lib/runAll.js#L116-L118

Reply:
Yeah I think you should catch those here as well?

Reply:
This lead to an insane rabbit hole because I totally forgot about having the global context where I can push errors, and handle them after the fact. 😅

So I'll refactor the chain of `getConfigGroups` -> `loadConfig` -> `validateConfig` -> `validateBraces` to use that instead.